src/extension/ExtensionContextHandler.ts

```diff
@@ -180,6 +181,14 @@ export class ExtensionContextHandler {
 		this.registerDeploymentsRouteCommands(deploymentsProvider);
 	}
 
+	public async getRuntimeInfo(): Promise<Runtime> {
+		const jbangExec = await getRuntimeInfo();
+
+		// Store the runtime in the context for later use
```

Comment:
This would need to be stored for each folder as it can be different for each of them. Theorically, this also can be modified (especially the version) while the workspace is opened but could be fine to require a restart of VS Code as a first iteration.

Reply:
Yes, it will be queried per file, not per workspace :+1:

Reply:
That comment will be deleted as well